### DISCUSSION TITLE
Interesting Pi projects

### DISCUSSION THREAD

**Discussion by andrewdavidmackenzie** [andrewdavidmackenzie/pigg#124] [General]
I thought that this Pi based Oscillascope was similar in part to some of the initial ideas (or needs) I had, that led to "pigg"...

https://bitscope.com/pi/

**Comment by mgrouch** [comment]
There is something similar in openplotter project 

![](https://openplotter.readthedocs.io/en/latest/_images/settings10.png)

https://forum.openmarine.net/attachment.php?aid=1999

https://github.com/openplotter/openplotter-gpio

You might take a look at that use case.

**Reply by andrewdavidmackenzie** [reply]
That looks interesting, especially anything related to plotting inputs over time - an issue I'm working on now.

As we want to stay based on rust, we may end up reimplementing things from the python ecosystem....

**Reply by andrewdavidmackenzie** [reply]
I took a quick look at open plotter.

From the screenshot and name I thought it was generic to plotting the GPIO input levels, but it turns out it's all about GPS positions and plotting positions (of a boat), using a GPS connected to a Pi.
Did I get that right?

Either way, interesting.

**Reply by mgrouch** [reply]
No it’s not about gps. It’s assigning gpio pins to different inputs and reading from them 
1-wire protocol (Dallas temperature sensors, pwm for engine RPM, bilge switch, etc, and generating sensors data feed into SignalK which is a separate marine data server running on it. 

You might do something more generic. Like feeding HomeAssistant

**Reply by andrewdavidmackenzie** [reply]
I couldn't see anything in the README in the GH repo about data types, so I went to the docs site:

https://openplotter.readthedocs.io/en/3.x.x/

And there it all seemed to be about GPS and boats (even an anchor icon), did I stop reading too soon?

I also notice the forum is an open marine one.

**Reply by andrewdavidmackenzie** [reply]
BTW, Interested to hear any ideas you have on home assistant integration or related features...

**Reply by mgrouch** [reply]
I looked via code of Openplotter gpio itself. 
another idea might be to make it more generic
why not to create a software and gui configurator to expose gpio sensors and their readings via MQTT?

**Reply by mgrouch** [reply]
For i2c I wrote i2c scanner in rust which also checks addresses and lists possible compatible devices for them

https://github.com/bareboat-necessities/rust-modules/tree/main/rust-i2c-scan

**Reply by andrewdavidmackenzie** [reply]
MQTT (or similar) for sending sensor readings is an interesting idea.
Want to take a shot at writing an issue describing it @mgrouch ?
